File: go/hamming/hamming.go
//Package hamming implements Hamming's distance
package hamming

import "errors"

//Distance calculate Hamming distance of 2 DNA strings
func Distance(a, b string) (int, error) {
	if len(a) != len(b) {
		return 0, errors.New("Err")
	}
	var sum int
	for i := 0; i < len(a); i++ {
		if a[i] != b[i] {
			sum++
		}
	}

	return sum, nil
}

File: r/hamming/hamming.R
# This is a stub function to take two strings
# and calculate the hamming distance
hamming <- function(strand1, strand2) {
  if(nchar(strand1) != nchar(strand2)) {
    stop("Error!")
  }
  bv = unlist(strsplit(strand1,"")) == unlist(strsplit(strand2,""))
  sum(!bv)
}

# hamming("A","A")
